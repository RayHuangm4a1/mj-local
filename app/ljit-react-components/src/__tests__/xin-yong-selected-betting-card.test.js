import React from 'react';
import { shallow, mount, } from 'enzyme';
import XinYongSelectedBettingCard from '../components/xing-yong-selected-betting-card';

describe('XinYongSelectedBettingCard', () => {
	it('handle default props', () => {
		const {
			data,
			onClose,
			isShowingCloseButton,
			className,
		} = XinYongSelectedBettingCard.defaultProps;

		expect(data).toEqual({});
		expect(onClose).toBeDefined();
		expect(onClose).toBeInstanceOf(Function);
		expect(isShowingCloseButton).toBe(true);
		expect(className).toBe('');
	});

	it('should renders correctly', () => {
		const wrapper = shallow(
			<XinYongSelectedBettingCard
				data={{
					betcontent: '小',
					amount: 100,
					odds: 1.996,
					reward: 200,
				}}
				className='mock-class'
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class selected-betting-card', () => {
		const wrapper = shallow(<XinYongSelectedBettingCard/>);

		expect(wrapper.hasClass('selected-betting-card')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<XinYongSelectedBettingCard className={className}/>);

		expect(wrapper.hasClass(className)).toEqual(true);
	});
	it('should handle onClose event', () => {
		const onClose = jest.fn();
		const wrapper = mount(<XinYongSelectedBettingCard onClose={onClose} />
		);

		wrapper.find('.ljit-info-card__close-button').simulate('click');
		expect(onClose).toHaveBeenCalledTimes(1);
		expect(onClose).toHaveBeenCalled();
	});

	it('should mount in a full DOM', () => {
		const data={
			betcontent: '小',
			amount: 100,
			odds: 1.996,
			reward: 200,
		};
		const className = 'mock-class';
		const onClose = () => {};
		const isShowingCloseButton = true;
		const wrapper = mount(
			<XinYongSelectedBettingCard
				className={className}
				data={data}
				onClose={onClose}
				isShowingCloseButton={isShowingCloseButton}
			/>
		);

		expect(wrapper.props().className).toBe(className);
		expect(wrapper.props().data).toMatchObject(data);
		expect(wrapper.props().onClose).toEqual(onClose);
		expect(wrapper.props().isShowingCloseButton).toEqual(isShowingCloseButton);
	});
});
