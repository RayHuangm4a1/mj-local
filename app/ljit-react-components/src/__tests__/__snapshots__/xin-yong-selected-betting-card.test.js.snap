// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XinYongSelectedBettingCard should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <XinYongSelectedBettingCard
    className="mock-class"
    data={
      Object {
        "amount": 100,
        "betcontent": "小",
        "odds": 1.996,
        "reward": 200,
      }
    }
    isShowingCloseButton={true}
    onClose={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "selected-betting-card mock-class",
      "isShowingCloseButton": true,
      "left": <React.Fragment>
        <LabelText
          className="selected-betting-card__number"
          fontSize="medium"
          label="号码 :"
          labelColType="large"
          labelStyle={Object {}}
          text="小"
          textStyle={Object {}}
        />
        <LabelText
          className="selected-betting-card__amount"
          fontSize="medium"
          label="金额 :"
          labelColType="large"
          labelStyle={Object {}}
          text={100}
          textStyle={Object {}}
        />
      </React.Fragment>,
      "onClose": [Function],
      "right": <React.Fragment>
        <LabelText
          fontSize="medium"
          label="赔率 :"
          labelColType="large"
          labelStyle={Object {}}
          text={
            <DecimalNumber
              className=""
              data={1.996}
              decimalPlaces={4}
              hasSeparator={true}
              isCurrency={false}
              isPercent={false}
              isTolerance={false}
            />
          }
          textStyle={Object {}}
        />
        <LabelText
          fontSize="medium"
          label="可赢 :"
          labelColType="large"
          labelStyle={Object {}}
          text={
            <DecimalNumber
              className=""
              data={200}
              decimalPlaces={4}
              hasSeparator={true}
              isCurrency={false}
              isPercent={false}
              isTolerance={false}
            />
          }
          textStyle={Object {}}
        />
      </React.Fragment>,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "selected-betting-card mock-class",
        "isShowingCloseButton": true,
        "left": <React.Fragment>
          <LabelText
            className="selected-betting-card__number"
            fontSize="medium"
            label="号码 :"
            labelColType="large"
            labelStyle={Object {}}
            text="小"
            textStyle={Object {}}
          />
          <LabelText
            className="selected-betting-card__amount"
            fontSize="medium"
            label="金额 :"
            labelColType="large"
            labelStyle={Object {}}
            text={100}
            textStyle={Object {}}
          />
        </React.Fragment>,
        "onClose": [Function],
        "right": <React.Fragment>
          <LabelText
            fontSize="medium"
            label="赔率 :"
            labelColType="large"
            labelStyle={Object {}}
            text={
              <DecimalNumber
                className=""
                data={1.996}
                decimalPlaces={4}
                hasSeparator={true}
                isCurrency={false}
                isPercent={false}
                isTolerance={false}
              />
            }
            textStyle={Object {}}
          />
          <LabelText
            fontSize="medium"
            label="可赢 :"
            labelColType="large"
            labelStyle={Object {}}
            text={
              <DecimalNumber
                className=""
                data={200}
                decimalPlaces={4}
                hasSeparator={true}
                isCurrency={false}
                isPercent={false}
                isTolerance={false}
              />
            }
            textStyle={Object {}}
          />
        </React.Fragment>,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
